src/backend/distributed/utils/shardinterval_utils.c

```diff
@@ -347,10 +347,22 @@ FindShardIntervalIndex(Datum searchedValue, CitusTableCacheEntry *cacheEntry)
 		}
 		else
 		{
-			int hashedValue = DatumGetInt32(searchedValue);
-			uint64 hashTokenIncrement = HASH_TOKEN_COUNT / shardCount;
+			/*
+			 * We use a 64-bit integer to be able to perform arithmetic
```

Comment:
Could be nice to have this else block in a function like `FindShardIndexFromValue`

Reply:
Fixed